main/inc/lib/myspace.lib.php

```diff
@@ -1156,137 +1156,253 @@ public static function displayResumeLP(
+            } else {
+                $items = explode(',', $authorData);
+                for ($j = 0; $j < count($items); $j++) {
+                    $authorData = $items[$j];
+                    if (!isset($users[$authorData])) {
+                        $users[$authorData] = api_get_user_info($authorData);
+                    }
+                    $teachers[$authorData][$lpId] = $users[$authorData];
+                    $learningPaths[$lpId]['teachers'][$authorData] = $users[$authorData];
                 }
             }
         }
-        if ($csv == false) {
-            $table = "<div class='table-responsive'>".
+        $lpItems = array_unique($lpItems);
+        $whereInLp = implode(',', $lpItems);
+        if (count($lpItems) != 0) {
+            $registeredUsers = self::getCompanyLearnpathSubscription(
+                $startDate,
+                $endDate,
+                $whereInLp);
+            foreach ($registeredUsers as $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
+                    $user = $student[$i];
+                    $lpId = $user['lp_item'];
+                    $studentId = $user['id'];
+                    if (!isset($users[$studentId])) {
+                        $users[$studentId] = api_get_user_info($studentId);
+                        $users[$studentId]['company'] = $user['company'];
+                    }
+                    $learningPaths[$lpId]['courseStudent'][$studentId] = $users[$studentId];
+                }
+            }
+
+            $registeredUsersBySession = self::getSessionAddUserCourseFromTrackDefault(
+                $startDate,
+                $endDate,
+                $whereInLp
+            );
+            foreach ($registeredUsersBySession as $lpId => $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
+                    $user = $student[$i];
+                    $lpId = $user['lp'];
+                    $studentId = $user['id'];
+                    if (!isset($users[$studentId])) {
+                        $users[$studentId] = api_get_user_info($studentId);
+                        $users[$studentId]['company'] = $user['company'];
+                    }
+                    $learningPaths[$lpId]['sessionStudent'][$studentId] = $users[$studentId];
+                    $learningPaths[$lpId]['sessionStudent'][$studentId]['session_id'] = $user['session_id'];
+                }
+            }
+
+            $registeredUsersGroup = self::getCompanyLearnpathSubscription(
+                $startDate,
+                $endDate,
+                $whereInLp,
+                true
+            );
+            foreach ($registeredUsersGroup as $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
```

Comment:
Mismo comentario que arriba

Reply:
comentario añadido "use 'for' to performance"